### ISSUE TITLE
Core ML nms=True' is not available for end2end models. Forcing 'nms=False'.

### ISSUE THREAD

**Issue by philipperemy** [ultralytics/ultralytics#16927]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Model Export

### Bug

Ultralytics 8.3.3 🚀 Python-3.11.6 torch-2.4.0 CPU (Apple M1 Max)
YOLOv10m summary (fused): 369 layers, 16,451,542 parameters, 0 gradients, 63.4 GFLOPs

PyTorch: starting from '/Users/premy/PycharmProjects/weeds-onions/models/hemp-yolov10-medium-20240708/weights.pt' with input shape (1, 3, 1024, 1024) BCHW and output shape(s) (1, 300, 6) (32.0 MB)

CoreML: starting export with coremltools 8.0...
CoreML: WARNING ⚠️ 'nms=True' is not available for end2end models. Forcing 'nms=False'.

Does someone know why? It's a YOLO v10 medium model for object detection with just one class.

**Does it mean all YOLO v10 can't be exported?**

### Environment

Python 3.11
Mac OS

### Minimal Reproducible Example

```python
model = YOLO(weights)
model_dir = model.export(format='mlpackage', nms=True)  # creates 'yolo11n.mlpackage'
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Y-T-G** [comment]
YOLOv10 doesn't use NMS.

**Comment by philipperemy** [comment]
@Y-T-G okay gotcha but do you know how can I get the preview with MacOS? I need NMS for that.

<img width="851" alt="image" src="https://github.com/user-attachments/assets/c0d0eded-f9e3-497a-9512-3d037c03b435">

**Comment by UltralyticsAssistant** [comment]
👋 Hello @philipperemy, thank you for reaching out and using Ultralytics 🚀!

For your query regarding the Core ML export issue where `'nms=True' is not available for end2end models`, it sounds like a concern with the integration between Core ML tools and YOLO. This is an automated response to guide you through some initial steps while an Ultralytics engineer also looks into your issue soon.

If this is a 🐛 Bug Report, thank you for already providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). It's helpful for diagnosing the problem more efficiently.

Here are some suggestions while we investigate:

1. Ensure you’re running the latest version of the `ultralytics` package and dependencies. You can upgrade with:

   ```bash
   pip install -U ultralytics
   ```

2. Double-check that you're using an appropriate environment. YOLO models should be run in environments that adhere to all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). It's also a good idea to test in different environments, such as [Google Colab](https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb) or a verified [Docker](https://hub.docker.com/r/ultralytics/ultralytics) setup.

3. Explore further documentation:
   - For exporting models, take a look at our [export guide](https://docs.ultralytics.com/export/).
   - For understanding non-maximum suppression in exports, this [Core ML section](https://docs.ultralytics.com/export/coreml/) might be useful.

Join our community channels for real-time discussions and support:
- [Discord](https://ultralytics.com/discord) 🎧
- [Discourse](https://community.ultralytics.com)
- [Subreddit](https://reddit.com/r/ultralytics)

Feel free to follow up or provide additional details as necessary. Your patience is appreciated while we work on finding a resolution!

**Comment by philipperemy** [comment]
@Y-T-G okay gotcha but do you know how can I get the preview with MacOS? I need NMS for that.

<img width="851" alt="image" src="https://github.com/user-attachments/assets/c0d0eded-f9e3-497a-9512-3d037c03b435">

**Comment by Y-T-G** [comment]
That's probably a CoreML issue. The model should still work if it works in `ultralytics`

**Comment by philipperemy** [comment]
yes it does work seamlessly with `ultralytics`. The issue is the export to `Core ML`. I will leave it open for this reason.

**Comment by Y-T-G** [comment]
You can run the Core ML model in `ultralytics` too. If it works, then there isn't an issue with export.

**Comment by DenDen047** [comment]
It appears that the ONNX conversion might have the same issue.

In the screenshot, the left side shows the ONNX model without the `nms=True` option, while the right side includes the `nms=True` option. The results seem to have the same architecture.
![Screenshot 2024-11-03 at 0 35 57](https://github.com/user-attachments/assets/bce0e9fa-6aac-4afd-84e1-07dc13d63d6d)

**Comment by Y-T-G** [comment]
@DenDen047 ONNX export in `ultralytics`  doesn't support `nms=True`. Only arguments listed in the table for the format is supported.

https://docs.ultralytics.com/modes/export/#arguments

**Comment by philipperemy** [comment]
Well, long story short, you'll have to implement the NMS yourself with Swift because it is no longer supported. It used to be up to Yolo v8 (if my memory is correct).

**Comment by glenn-jocher** [comment]
Thank you for your input. Implementing NMS in Swift is indeed a viable solution for handling post-processing in Core ML models. If you need further assistance, feel free to ask.

**Comment by philipperemy** [comment]
Well ideally it should be part of the ultralytics framework.

**Comment by glenn-jocher** [comment]
Thank you for the suggestion. While NMS functionality for exported models is not currently integrated, it remains a feature request for potential future development. Contributions via PRs are always welcome!

**Comment by Y-T-G** [comment]
`nms` is supported for CoreMLTool. YOLOv10 doesn't use or need NMS, so it doesn't export with `nms`. YOLO11 is similar to YOLOv8 and also uses NMS.

**Comment by Y-T-G** [comment]
For preview, maybe try exporting with `format="mlmodel"`
